tests/programs/dlopen_demo.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
```

Comment:
Note that we have a program `run_demos.py`, invoked by `all.sh` components, which runs all the scripts called `*_demo.sh`. Please make sure to update this script so that it looks in all `framework/tests/programs`.

On a related note, [`run_demos.py` should move to the framework](https://github.com/Mbed-TLS/mbedtls-framework/issues/87). It may be easier to do this before changing it.

Reply:
Reviewers beware: there's no automatic mechanism that would let us know if we stop running the demo programs on the CI. They aren't even recorded in the outcome file. So please check the CI logs.

Reply:
Good point, thanks. However, currently,  dlopen_demo.sh is actually never run through run_demos.py as there is no component where shared libraries are built and run_demos.py is run. dlopen_demo.sh is run directly in `test_make_shared` and `test_cmake_shared`.

Reply:
This PR is already quite big, I prefer do that rather as part of https://github.com/Mbed-TLS/TF-PSA-Crypto/issues/127. I've added a note there to not forget about it.